libs/api/domains/icelandic-government-institution-vacancies/src/lib/utils.ts

```diff
@@ -219,12 +125,53 @@ export const mapIcelandicGovernmentInstitutionVacanciesFromExternalSystem =
 
 export const mapIcelandicGovernmentInstitutionVacancyByIdResponseFromExternalSystem =
   async (
-    vacancy: DefaultApiVacancyDetails,
+    vacancy: VacancyResponseDto,
   ): Promise<IcelandicGovernmentInstitutionVacancyByIdResponse['vacancy']> => {
-    const item = vacancy.starfsauglysing
+    const locations: IcelandicGovernmentInstitutionVacanciesResponse['vacancies'][number]['locations'] =
+      []
 
-    const contacts = mapContacts(item)
-    const locations = mapLocations(item)
+    const locationTitles =
+      vacancy.locations
+        ?.split(',')
+        .map((location) => location.trim())
+        .filter(Boolean) ?? []
+
+    for (const title of locationTitles) {
+      locations.push({
+        postalCode: vacancy.postCode ?? undefined,
+        title,
+      })
+    }
+
+    const contacts: IcelandicGovernmentInstitutionVacancyContact[] = []
+
+    if (
+      vacancy.contact1Name ||
```

Comment:
Perhaps check the trimmed version of the fields, I noticed on dev there are sometimes " " values.